internal/gatewayapi/filters.go

```diff
@@ -1065,3 +1073,12 @@ func (t *Translator) processInvalidHTTPFilter(filterType string, filterContext *
 		StatusCode: ptr.To(uint32(500)),
 	}
 }
+
+// TODO (liorlieberman) is there a better home for this func?
+// OrDefault returns *t if its non-nil, or else def.
+func OrDefault[T any](t *T, def T) T {
```

Comment:
can you use existing ptr.Deref() instead

Reply:
sure, thanks for pointing it out